lib/omniauth/strategies/shopify.rb

```diff
@@ -26,7 +26,11 @@ class Shopify < OmniAuth::Strategies::OAuth2
       option :setup, proc { |env|
         strategy = env['omniauth.strategy']
 
-        shopify_auth_params = strategy.session['shopify.omniauth_params'] && strategy.session['shopify.omniauth_params'].with_indifferent_access
+        shopify_auth_params = strategy.session['shopify.omniauth_params'] ||
+          strategy.session['omniauth.params'] ||
```

Comment:
Why is `with_indifferent_access` removed here?

Reply:
moved here https://github.com/Shopify/omniauth-shopify-oauth2/pull/91/files#diff-20fb14b92659da69e72effc9ab8f72feR33 to prevent duplicated call `with_indifferent_access` for each of "params place"

I found only 3 places `shop` param come from:
`strategy.session['shopify.omniauth_params']` - set by shopify app gem
`strategy.session['omniauth.params']` - omniauth sets this key in request phase(https://github.com/omniauth/omniauth/blob/master/lib/omniauth/strategy.rb#L208) and deletes it before callback. But setup phase runs before delete. So we can use this key on callback phase
`strategy.request.params` - get shop from params